mlflow/models/evaluation/base.py

```diff
@@ -1178,9 +1182,17 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
 
 def _convert_data_to_mlflow_dataset(data, targets=None, predictions=None):
     """Convert input data to mlflow dataset."""
+    supported_dataframe_types = [pd.DataFrame]
     if "pyspark" in sys.modules:
         from pyspark.sql import DataFrame as SparkDataFrame
 
+        supported_dataframe_types.append(SparkDataFrame)
+
+    if predictions is not None:
+        _validate_dataset_type_supports_predictions(
+            data=data, supported_predictions_dataset_types=supported_dataframe_types
+        )
```

Comment:
We removed the upstream assertion that `predictions` must be a pandas dataframe when `model` is unspecified: https://github.com/mlflow/mlflow/pull/11831/files#r1580245904. 

We need to perform this check before creating the `EvaluationDataset`, else the predictions specification from the user might be discarded altogether if the dataset is not a DataFrame. The user would get a confusing error like "predictions must be specified", even if they actually *did* specify predictions. This statement ensures that the user gets the *correct* error: "predictions" can't be specified with non-DataFrame types.